File: LeetCode/Algo/DP/climb_stairs.java
import java.util.Scanner;
class Solution {
    public int climbStairs(int n) {
        if(n==1) return 1;
        if(n==0) return 0;
        int pre = 1;
        int curr = 2;
        for(int i = 3; i <= n; i++){
            int sum = pre + curr;
            pre = curr;
            curr = sum;
        }
        return curr;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Solution solve = new Solution();
        System.out.println(solve.climbStairs(n));
    }
}

File: LeetCode/Algo/DP/climb_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        pre = 1 # n = 1
        curr = 2 # n = 2
        for i in range(3, n+1):
            sum = pre + curr
            pre = curr
            curr = sum 
        return curr
            
solve = Solution()
print(solve.climbStairs(3))

# bài này kiểu cho số n , tìm số cách để tống lại bằng n 
# các số chỉ có thể là 1, 2

# 2 => 1 + 1 , 2
# 3 => 1 + 1 + 1, 1 + 2, 2 + 1
# 1 1 2 3 5 8 13 